core/lib/dal/src/fri_proof_compressor_dal.rs

```diff
@@ -89,8 +89,22 @@ impl FriProofCompressorDal<'_, '_> {
         .fetch_optional(self.storage.conn())
         .await
         .unwrap()
-        .map(|row| L1BatchNumber(row.l1_batch_number as u32));
-        result
+        .map(|row| L1BatchNumber(row.l1_batch_number as u32))
+    }
+
+    pub async fn get_proof_compression_job_attempts(
+        &mut self,
+        l1_batch_number: L1BatchNumber,
+    ) -> Option<u32> {
```

Comment:
Maybe, it makes sense to propagate `sqlx` errors right away for new methods (as in `get_basic_witness_input_producer_job_attempts`)?

Reply:
Agree